staging/src/k8s.io/apimachinery/pkg/util/httpstream/httpstream.go

```diff
@@ -42,10 +42,9 @@ func NoOpNewStreamHandler(stream Stream, replySent <-chan struct{}) error { retu
 
 // Dialer knows how to open a streaming connection to a server.
 type Dialer interface {
-
 	// Dial opens a streaming connection to a server using one of the protocols
 	// specified (in order of most preferred to least preferred).
-	Dial(protocols ...string) (Connection, string, error)
+	Dial(remotePorts []uint16, protocols ...string) (Connection, string, error)
```

Comment:
this interface isn't specific to portforward... I don't think adding parameters specific to portforwarding makes sense here